server/build_event_protocol/target_tracker/target_tracker.go

```diff
@@ -234,7 +234,7 @@ func (t *TargetTracker) writeTestTargetStatuses(ctx context.Context, permissions
 	}
 	newTargetStatuses := make([]*tables.TargetStatus, 0)
 	for _, target := range t.targets {
-		if !isTest(target) {
+		if !isTest(target) || target.overallStatus == build_event_stream.TestStatus_NO_STATUS {
 			continue
 		}
```

Comment:
Discussed offline. One benefit about writing no status here is that if there are cases where the test failed and we mislabeled them as NO_STATUS (for example, "ABORTED" or "FAILED TO BUILD". It will be easier to find them compared to they don't appear in the db in the first place.